bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoCrockpotHandler.java

```diff
@@ -0,0 +1,323 @@
+
+        this.wemoHttpCaller = wemohttpCaller;
+
+        logger.debug("Creating a WemoCrockpotHandler for thing '{}'", getThing().getUID());
+
+        if (upnpIOService != null) {
+            this.service = upnpIOService;
+        } else {
+            logger.debug("upnpIOService not set.");
+        }
+    }
+
+    @Override
+    public void initialize() {
+        Configuration configuration = getConfig();
+
+        if (configuration.get("udn") != null) {
+            logger.debug("Initializing WemoCrockpotHandler for UDN '{}'", configuration.get("udn"));
+            service.registerParticipant(this);
+            onSubscription();
+            onUpdate();
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            logger.debug("Cannot initalize WemoCrockpotHandler. UDN not set.");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("WeMoCrockpotHandler disposed.");
+
+        removeSubscription();
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.trace("Command '{}' received for channel '{}'", command, channelUID);
+        String mode = "0";
+        String time = null;
+
+        if (command instanceof RefreshType) {
+            try {
+                updateWemoState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        } else if (channelUID.getId().equals(CHANNEL_COOKMODE)) {
+            String commandString = command.toString();
+            switch (commandString) {
+                case "OFF":
+                    mode = "0";
+                    time = "0";
+                    break;
+                case "WARM":
+                    mode = "50";
+                    break;
+                case "LOW":
+                    mode = "51";
+                    break;
+                case "HIGH":
+                    mode = "52";
+                    break;
+            }
+            try {
+                String soapHeader = "\"urn:Belkin:service:basicevent:1#SetBinaryState\"";
+                String content = "<?xml version=\"1.0\"?>"
+                        + "<s:Envelope xmlns:s=\"http://schemas.xmlsoap.org/soap/envelope/\" s:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+                        + "<s:Body>" + "<u:SetCrockpotState xmlns:u=\"urn:Belkin:service:basicevent:1\">" + "<mode>"
+                        + mode + "</mode>" + "<time>" + time + "</time>" + "</u:SetCrockpotState>" + "</s:Body>"
+                        + "</s:Envelope>";
+                String wemoURL = getWemoURL("basicevent");
+
+                if (wemoURL != null) {
+                    wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
+                }
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
See above

Reply:
Changed to RuntimeException, as Eclipse says there cannot be an IOException.